deployment/terraform/keycloak.go

```diff
@@ -0,0 +1,279 @@
+package terraform
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/mattermost/mattermost-load-test-ng/deployment"
+	"github.com/mattermost/mattermost-load-test-ng/deployment/terraform/assets"
+	"github.com/mattermost/mattermost-load-test-ng/deployment/terraform/ssh"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+)
+
+func (t *Terraform) setupKeycloak(extAgent *ssh.ExtAgent) error {
+	keycloakBinPath := "/opt/keycloak/keycloak-" + t.config.ExternalAuthProviderSettings.KeycloakVersion + "/bin"
+
+	mlog.Info("Configuring keycloak", mlog.String("host", t.output.KeycloakServer.PublicIP))
+
+	command := "start-dev"
+	extraArguments := []string{}
+
+	if !t.config.ExternalAuthProviderSettings.DevelopmentMode {
+		command = "start"
+	}
+
+	sshc, err := extAgent.NewClient(t.output.KeycloakServer.PublicIP)
+	if err != nil {
+		return err
+	}
+
+	// Check if the keycloak database exists
+	result, _ := sshc.RunCommand(`sudo -iu postgres psql -l | grep keycloak 2> /dev/null`)
+	if len(result) == 0 {
+		mlog.Info("keycloak database not found, creating it and associated role")
+		_, err = sshc.RunCommand(`sudo -iu postgres psql <<EOSQL
+		CREATE USER keycloak WITH PASSWORD 'mmpass';
+		CREATE DATABASE keycloak OWNER keycloak;
+		GRANT ALL PRIVILEGES ON DATABASE keycloak TO keycloak;
```

Comment:
Please move this to a .sql file under assets, copy it to the keycloak node via provisioner like:

```
provisioner "file" {
    source      = var.mattermost_license_file
    destination = "/home/ubuntu/mattermost.mattermost-license"
  }
```

And use `psql -f` to execute from the file.

Reply:
Mhmmm, how do I keep this in assets and then assign it to a variable if it's bundled in the binary?